File: src/main/scala/eventstore/javaexamples/ReadEventExample.java
package eventstore.javaexamples;

import akka.actor.*;
import akka.actor.Status.Failure;
import akka.event.*;
import eventstore.*;
import eventstore.j.*;
import eventstore.tcp.ConnectionActor;

import java.net.InetSocketAddress;


public class ReadEventExample {

    public static void main(String[] args) {
        final ActorSystem system = ActorSystem.create();
        final Settings settings = new SettingsBuilder()
                .address(new InetSocketAddress("127.0.0.1", 1113))
                .defaultCredentials("admin", "changeit")
                .requireMaster(true)
                .build();
        final ActorRef connection = system.actorOf(Props.create(ConnectionActor.class, settings));
        final ActorRef readEventActor = system.actorOf(Props.create(ReadEventActor.class, connection));
    }


    public static class ReadEventActor extends UntypedActor {
        final LoggingAdapter log = Logging.getLogger(getContext().system(), this);

        ActorRef connection;

        public ReadEventActor(ActorRef connection) {
            this.connection = connection;
        }

        @Override
        public void preStart() throws Exception {
            final ReadEvent readEvent = new ReadEventBuilder("my-stream")
                    .eventNumberFirst()
                    .resolveLinkTos(false)
                    .requireMaster(true)
                    .build();

            connection.tell(readEvent, getSelf());
        }

        public void onReceive(Object message) throws Exception {
            if (message instanceof ReadEventCompleted) {
                final ReadEventCompleted completed = (ReadEventCompleted) message;
                final Event event = completed.event();
                log.info("EVENT: " + event.toString());
            } else if (message instanceof Failure) {
                final Failure failure = ((Failure) message);
                final EventStoreException exception = (EventStoreException) failure.cause();
                log.error("FAILED: reason: {}, message: {}", exception.reason(), exception.message());
            } else
                unhandled(message);
        }
    }
}

File: examples/src/main/scala/eventstore/akka/examples/ReadEventExample.scala
package eventstore
package akka
package examples

import java.net.InetSocketAddress
import _root_.akka.actor._
import _root_.akka.actor.Status.Failure
import eventstore.akka.tcp.ConnectionActor

object ReadEventExample extends App {
  val system = ActorSystem()

  val settings = Settings(
    address = new InetSocketAddress("127.0.0.1", 1113),
    defaultCredentials = Some(UserCredentials("admin", "changeit"))
  )

  val connection = system.actorOf(ConnectionActor.props(settings))
  implicit val readResult = system.actorOf(Props[ReadResult])

  connection ! ReadEvent(EventStream.Id("my-stream"), EventNumber.First)

  class ReadResult extends Actor with ActorLogging {
    def receive = {
      case ReadEventCompleted(event) =>
        log.info("event: {}", event)
        context.system.terminate()

      case Failure(e: EsException) =>
        log.error(e.toString)
        context.system.terminate()
    }
  }
}